agent/task.go

```diff
@@ -207,10 +207,15 @@ func (tc *taskContext) setCurrentIdleTimeout(cmd command.Command) {
 	tc.Lock()
 	defer tc.Unlock()
 
-	var timeout time.Duration
 	if cmd == nil {
-		timeout = defaultIdleTimeout
-	} else if dynamicTimeout := tc.taskConfig.GetIdleTimeout(); dynamicTimeout != 0 {
+		tc.setIdleTimeout(defaultIdleTimeout)
+		tc.logger.Execution().Debugf("Set idle timeout for %s (%s) to %s.",
+			tc.currentCommand.DisplayName(), tc.currentCommand.Type(), tc.getIdleTimeout())
+		return
```

Comment:
As described in the above comment, we don't need to re-add the nil check as long as we always pass in the current command (i.e. remove the if-else conditional above and just `tc.setCurrentIdleTimeout`.